.github/workflows/check_compatibility_icy.yaml

```diff
@@ -0,0 +1,105 @@
+      run: pip install requests==2.28.2 beautifulsoup4==4.12.3 packaging==23.0
+    - name: Generate dict
+      id: set-dict
+      run: |
+        matrix_output=$(python scripts/get_java_software_versions.py icy)
+        echo "matrix=${matrix_output}" >> $GITHUB_OUTPUT
+    - name: Versions matrix
+      run: echo '${{ steps.set-dict.outputs.matrix }}'
+
+  run:
+    needs: generate-version-dict
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.generate-version-dict.outputs.matrix) }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/checkout@v3
+      with:
+        repository: bioimage-io/collection-bioimage-io
+        ref: gh-pages
+        path: bioimageio-gh-pages
+    - name: Setup Maven Action
+      uses: stCarolas/setup-maven@v4.5
+      with:
+        java-version: 11
+        maven-version: 3.9.5
+    - name: Build with Maven
+      run: |
+        cd scripts/check_compatibility_java_software
+        mvn clean install org.apache.maven.plugins:maven-shade-plugin:3.2.4:shade -Dshade.mainClass=io.bioimage.modelrunner.ci.ContinuousIntegration
+    - name: Get jar file name
+      id: get-jar
+      run: |
+        cd scripts/check_compatibility_java_software
+        jarfile=$(mvn -q exec:exec -Dexec.executable=echo -Dexec.args='${project.build.finalName}.jar')
+        if [ -z "$jarfile" ]; then
+          echo "Failed to get jar file name"
+          exit 1
+        fi
+        if [ ! -f "target/$jarfile" ]; then
+          echo "Jar file not found: target/$jarfile"
+          exit 1
+        fi
+        echo "Jar file found: target/$jarfile"
+        echo "jarfile=$jarfile" >> $GITHUB_OUTPUT
+    - name: Download engines
+      run: |
+        java -cp "scripts/check_compatibility_java_software/target/${{ steps.get-jar.outputs.jarfile }}" io.bioimage.modelrunner.ci.DownloadEngines icy
+    - name: Run models
+      run: |
+        java -cp "scripts/check_compatibility_java_software/target/${{ steps.get-jar.outputs.jarfile }}" io.bioimage.modelrunner.ci.ContinuousIntegration icy ${{ matrix.key }}
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        path: test_summaries_icy_${{ matrix.key }}
+
+  sendreport:
+    runs-on: ubuntu-latest
+    needs: [generate-version-dict, run]
+    strategy:
+      matrix: ${{ fromJson(needs.generate-version-dict.outputs.matrix) }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10'
+    - name: Get test results
+      uses: actions/download-artifact@v3
+      with:
+        path: test_summaries_icy_${{ matrix.key }}
+    - name: Install deps
+      run: |
+        pip install .
+    - name: Send icy ${{ matrix.key }} tests
+      shell: bash -l {0}
+      run: python scripts/check_compatibility_java_software.py icy ${{ matrix.key }} --summaries_dir "test_summaries_icy_${{ matrix.key }}/artifact"
```

Comment:
could you maybe reuse https://github.com/bioimage-io/collection/blob/main/scripts/upload_reports.py here instead?